app/views/course/assessment/answer/programming/_test_cases.html.slim

```diff
@@ -1,5 +1,5 @@
 - auto_grading = answer.auto_grading ? answer.auto_grading.actable : nil
-- is_grader = can?(:grade, answer.answer)
+- is_grader = can?(:read_tests, answer.answer)
```

Comment:
How about use an assessment or submission level permission instead ? 

Otherwise it means for each answer, we will fire queries to check `answer.answer.submission.assessment.tab.category.course.course_users`. Although most of the queries are cached, I think it's still better to avoid that.